### PR TITLE
Feature/story 16/logout

### PR DESCRIPTION
**Pull request by hale4029**
**Functionality:**
Logout link added to navbar and functionality created.

**Testing:**
Created -- flash message working and cart emptied. 

**Related Issues:**
May need to add additional visit page to prove logout worked -- regular user would need something similar to the 404 error

### CODE REVIEW

File: spec/features/users/logout_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'Logging in', type: :feature do
```
Comment (mintona) [inline spec/features/users/logout_spec.rb]: I think this should say "logging out" instead of logging in

File: spec/features/users/logout_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    it 'can login with valid credentials' do
+      mike = Merchant.create(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
+      paper = mike.items.create(name: "Lined Paper", description: "Great for writing on!", price: 20, image: "https://cdn.vertex42.com/WordTemplates/images/printable-lined-paper-wide-ruled.png", inventory: 25)
+      user_1 = create(:user, role: 0)
+
+      visit '/'
+      click_link('Login')
+      fill_in :email, with: user_1.email
+      fill_in :password, with: 'password1'
+      click_on('Submit')
```
Comment (mintona) [inline spec/features/users/logout_spec.rb]: This login process can be stubbed with   `allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user_1)`

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +0,0 @@
-<h1>Hello, <%= @user.name %>!</h1>
```
Comment (mintona) [inline app/views/users/show.html.erb]: Why is this file showing as deleted?

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,27 +0,0 @@
-class UsersController < ApplicationController
```
Comment (mintona) [inline app/controllers/users_controller.rb]: Why is this file showing as deleted?

**Review by mintona — CHANGES_REQUESTED** [review]
It looks like the functionality is working but I left comments about the two files that need to be deleted. 

Also left a comment about stubbing (mocking?) in a logged in user in the test, and one with what I believe to be a typo. 

I clicked the "request changes" circle in the review so let's see how that works :)

File: spec/features/users/logout_spec.rb (hunk 2 above)
Comment (hale4029) [reply spec/features/users/logout_spec.rb]: In order to create real session(s), it looks like we need to go through the motions so sessions are created and destroyed.

**Comment by mintona** [comment]
OK, I think I get what you're saying now.

**Review by hale4029 — COMMENTED** [review]
Reviewed.

**Review by mintona — APPROVED** [review]

**Review by mintona — APPROVED** [review]

**Comment by mintona** [comment]
Add changes requested look good.

**Pull request merged**